zulipterminal/ui_tools/buttons.py

```diff
@@ -15,25 +15,31 @@
 from zulipterminal.urwid_types import urwid_Size
 
 
+urwidMarkupTuple = Tuple[Optional[str], str]
+
+
 class TopButton(urwid.Button):
     def __init__(
         self,
         *,
         controller: Any,
-        caption: str,
+        prefix_markup: urwidMarkupTuple = (None, ""),
```

Comment:
Technically, it can also use a string as a prefix. I wanted to completely separate out TopButton from everything else in ZT. 
It is now a generic button with a flexible (more than a char) prefix and suffix. The suffix also has integer capabilities through count (can be extended to prefix if needed). So any ZT button can use however it needs.

Reply:
Also, wide unicode (emoji) prefixes might not fit if we force `len = 1`

Reply:
Right, the layout code has changed :+1: Originally the general button was purely to extract/abstract the common functionality, and we always used 0 or 1 characters as prefix.

We can use wider characters or multiple characters now at this base level with the new layout code, though it may be useful to coordinate this in the higher-level buttons? For example, everything will be offset if we use one character which is a different width, in a list of the same button type?

Reply:
We could create a subclass of Topbutton which forces `len = 1` and use that?
Though we do force the prefixes to be a char currently, either through hardcoding or basic logic for resolved topic.
Maybe we should wait for a double length prefix to show up before we do that refactor?

Reply:
For this PR I'm fine without, but we could enforce it in the subclasses we already use, where appropriate, eg. checking the actual length of the string/character argument?